app/src/main/java/com/github/wnder/GuessPreviewActivity.java

```diff
@@ -125,26 +126,21 @@ protected void onStart() {
         user = GlobalUser.getUser();
 
         //Get a new picture to display
-        try {
-            user.onNewPictureAvailable((LocationManager)getSystemService(Context.LOCATION_SERVICE), this, (picId) -> {
-                if(!picId.equals("")){
-                    //If there is a picture, display it
-                    picturesDb.getBitmap(picId).thenAccept((bmp) -> setImageViewBitmap(bmp, picId));
-                    picturesDb.getLocation(picId).thenAccept((Lct) -> {
-                        pictureLat = Lct.getLatitude();
-                        pictureLng = Lct.getLongitude();
-                    });
-                } else {
-                    //If not, display default picture
-                    // Maybe create a bitmap that tells that no pictures were available (this one is just the one available)
-                    Bitmap bmp = BitmapFactory.decodeResource(getResources(), R.raw.no_image);
-                    setImageViewBitmap(bmp, picId);
-                }
+        userDb.getNewPictureForUser(user).thenAccept(picId ->{
+            if(!picId.equals("")){
+                //If there is a picture, display it
+                picturesDb.getBitmap(picId).thenAccept((bmp) -> setImageViewBitmap(bmp, picId));
+                picturesDb.getLocation(picId).thenAccept((Lct) -> {
+                    pictureLat = Lct.getLatitude();
+                    pictureLng = Lct.getLongitude();
+                });
+            } else {
+                //If not, display default picture
+                // Maybe create a bitmap that tells that no pictures were available (this one is just the one available)
+                Bitmap bmp = BitmapFactory.decodeResource(getResources(), R.raw.no_image);
+                setImageViewBitmap(bmp, picId);
             }
-            );
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
+        });
```

Comment:
You could use the occassion to properly handle the exception for `getNewPictureForUser` (with .exceptionnaly) for example showing the "no image" picture.